pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -2748,15 +2781,62 @@ func TestSetKieServerFromBuild(t *testing.T) {
 								Name: byeRules,
 							},
 						},
-						Build: &api.KieAppBuildObject{},
+						Build: &api.KieAppBuildObject{
+							GitSource: api.GitSource{
+								URI: "https://test",
+							},
+						},
 					},
 				},
 			},
 		},
 	}
 	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting trial environment")
+	assert.Equal(t, helloRules, env.Servers[0].DeploymentConfigs[0].Spec.Triggers[0].ImageChangeParams.From.Name)
+	assert.Equal(t, "", env.Servers[0].DeploymentConfigs[0].Spec.Triggers[0].ImageChangeParams.From.Namespace)
+	assert.Equal(t, cr.Status.Applied.Objects.Servers[1].Name+latestTag, env.Servers[1].DeploymentConfigs[0].Spec.Triggers[0].ImageChangeParams.From.Name)
+	assert.Equal(t, "", env.Servers[1].DeploymentConfigs[0].Spec.Triggers[0].ImageChangeParams.From.Namespace)
+}
 
+func TestSetKieServerFromBuildAndWithImageTags(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment:  api.RhpamProductionImmutable,
+			UseImageTags: true,
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{
+					{
+						From: &api.ImageObjRef{
+							Kind: "ImageStreamTag",
+							ObjectReference: api.ObjectReference{
+								Name: helloRules,
+							},
+						},
+					},
+					{
+						From: &api.ImageObjRef{
+							Kind: "ImageStreamTag",
+							ObjectReference: api.ObjectReference{
+								Name: byeRules,
+							},
+						},
+						Build: &api.KieAppBuildObject{
+							GitSource: api.GitSource{
+								URI: "https://test",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	env, err := GetEnvironment(cr, test.MockService())
 	assert.Nil(t, err, "Error getting trial environment")
+
```

Comment:
any reason this check is missing in the new test?
`	assert.Equal(t, "", env.Servers[1].DeploymentConfigs[0].Spec.Triggers[0].ImageChangeParams.From.Namespace)`